File: server/db/models/spell.js
const mongoose = require('mongoose')

const spellSchema = mongoose.Schema({
  school: { type: String, default: '' },
  name: { type: String, default: '' },
  level: { type: String, default: '' },
  ee: { type: String, default: '' },
  ritual: { type: String, default: '' },
  casting_time: { type: String, default: '' },
  scag: { type: String, default: '' },
  id: { type: String, default: '' },
  source: { type: String, default: '' },
  range: { type: String, default: '' },
  classes: [ { type: String } ],
  components: { type: String, default: '' },
  duration: { type: String, default: '' },
  athigherlevel: { type: String, default: '' },
  concentration: { type: String, default: '' },
  slug: { type: String, default: '' },
  page: { type: String, default: '' },
  description: { type: String, default: '' }
})

module.exports = mongoose.model('Spell', spellSchema)

File: go-server/src/server/models/spell.go
package models

type (
	// Spell describes a spell
	Spell struct {
		Name          string   `json:"name,omitempty"`
		Level         string   `json:"level,omitempty"`
		School        string   `json:"school,omitempty"`
		Classes       []string `json:"classes,omitempty"`
		AtHigherLevel string   `json:"athigherlevel,omitempty"`
		CastingTime   string   `json:"casting_time,omitempty"`
		Components    string   `json:"components,omitempty"`
		Concentration string   `json:"concentration,omitempty"`
		Description   string   `json:"description,omitempty"`
		Duration      string   `json:"duration,omitempty"`
		EE            string   `json:"ee,omitempty"`
		ID            string   `json:"id,omitempty"`
		Page          string   `json:"page,omitempty"`
		Range         string   `json:"range,omitempty"`
		Ritual        string   `json:"ritual,omitempty"`
		Scag          string   `json:"scag,omitempty"`
		Slug          string   `json:"slug,omitempty"`
		Source        string   `json:"source,omitempty"`
		UniqueID      string   `json:"_id,omitempty"`
	}

	// Spells is an array of Spell structs
	Spells []Spell
)
